### ISSUE TITLE
X Forwarding doesn't work properly

### ISSUE THREAD

**Issue by marcinbojko** [Eugeny/terminus#2003]
Version: 1.0.99
Platform: linux 5.0.0-37-generic
Plugins: altair, clickable-links, save-output, scrollbar, shell-selector, theme-gruvbox, title-control

```
[SSH] Could not connect to the X server :6000: Error: connect ECONNREFUSED 127.0.0.1:6000
Failed to open connection to "session" message bus: /usr/bin/dbus-launch terminated abnormally with the following error: Autolaunch error: X11 initialization failed.

Running without a11y support!

[SSH] Could not connect to the X server :6000: Error: connect ECONNREFUSED 127.0.0.1:6000

[SSH] Could not connect to the X server :6000: Error: connect ECONNREFUSED 127.0.0.1:6000
Error: cannot open display: localhost:10.0
```

```
echo $DISPLAY
localhost:10.0
```

Do we need to do extra steps? Tested with other terminals without a problem

**Comment by Eugeny** [comment]
What is your distro?
Is this the $DISPLAY from the local side or on the remote side?

**Comment by marcinbojko** [comment]
Linux Mint 19.3.
I can check on any other Mint 19.x distro
$DISPLAY is remote side

**Comment by Eugeny** [comment]
What is the `DISPLAY` value on the local side?

**Comment by marcinbojko** [comment]
:0

**Comment by ohmyhub** [comment]
@marcinbojko This is what worked for me:

In the remote sshd_config set `X11UseLocalhost no`

Don't set `DISPLAY` manually in your .bashrc on the remote side.

I have the client-side `DISPLAY` value set to 0.0 as well.

That seemed to finally get things working for me.

However, I'm using Windows 10 to ssh into my Ubuntu VPS.

**Comment by marcinbojko** [comment]
@ohmyhub Thank you, but I'd prefer to have it working out of the box, beside voodoo. Steps are known to me, but these are artificial.

**Comment by XenHat** [comment]
> 
> 
> @ohmyhub Thank you, but I'd prefer to have it working out of the box, beside voodoo. Steps are known to me, but these are artificial.

For security reasons, many "Extra" features are disabled in SSH server by default. There's no _voodoo_ involved, simply knowing what you are changing, and why. It would be a security risk to let a remote connection modify system configuration.

**Comment by XenHat** [comment]
update: I also had the port 6000 error.
After installing VCXsrv (a X11 server) on my windows machine (via chocolatey), I confirm that X11 forwarding works as intended. Make sure your machine has a mean to accept the X11 connection.

I by no mean imply that it is an enjoyable experience (Although some config tweaks definitely do help), but I can run firefox remotely:
![image](https://user-images.githubusercontent.com/980408/78830109-6e03b680-79b5-11ea-9144-d86f5280d472.png)

**Comment by marcinbojko** [comment]
> update: I also had the port 6000 error.
> After installing VCXsrv (a X11 server) on my windows machine (via chocolatey), I confirm that X11 forwarding works as intended. Make sure your machine has a mean to accept the X11 connection.
> 
> I by no mean imply that it is an enjoyable experience (Although some config tweaks definitely do help), but I can run firefox remotely:
> ![image](https://user-images.githubusercontent.com/980408/78830109-6e03b680-79b5-11ea-9144-d86f5280d472.png)

Let me rephrase - this is not the best approach to change other endpoints setting - especially these you have no influence on. This is completely wrong solution saying: oh, yes, please do tweak your sshd.
I many cases it is not MY  sshd but managed by more mature organizations with different security level approach. Forcing them to introduce changes because 'my terminal app cannot do this' is just ... laughable.
Can I tweak my endpoint? One? Maybe. 500 ? Not gonna.

**Comment by gerritsj** [comment]
Version: 1.0.120
Platform: win32 10.0.17134
Plugins: none

Connecting to one remote machine with X11 forwarding works, but starting an X-application gives:
SSH   X  Could not connect to the X server :6000: Error: connect ECONNREFUSED 127.0.0.1:6000
FXApp::openDisplay: unable to open display localhost:11.0

Connecting to another remote machine with X11 forwarding:
SSH   X  Remote rejected opening a shell channel: Error: Unable to request X11

I have no privileges on these remote machines, so I can't change ssh configurations there.

I have no problems connecting to these remote machines with MobaXTerm.

**Comment by hqabcxyxz** [comment]
> Version: 1.0.120
> Platform: win32 10.0.17134
> Plugins: none
> 
> Connecting to one remote machine with X11 forwarding works, but starting an X-application gives:
> SSH X Could not connect to the X server :6000: Error: connect ECONNREFUSED 127.0.0.1:6000
> FXApp::openDisplay: unable to open display localhost:11.0
> 
> Connecting to another remote machine with X11 forwarding:
> SSH X Remote rejected opening a shell channel: Error: Unable to request X11
> 
> I have no privileges on these remote machines, so I can't change ssh configurations there.
> 
> I have no problems connecting to these remote machines with MobaXTerm.

I have a same problem like you,do you fix this problem now?

**Comment by rdricco** [comment]
Same here. Everything is fine on server side, I can connect using ssh -X but terminus option don't work.

**Comment by Eugeny** [comment]
Side note for anyone getting `ECONNREFUSED 127.0.0.1:6000` - you need a X server (vcxsrv, xming or similar) on your client. I'll be adding more useful help messages in the future.
